src/XamlStyler.dotUltimate/build.gradle

```diff
@@ -0,0 +1,155 @@
+
+sourceSets {
+    main {
+        java.srcDir 'src/rider/main/kotlin'
+        resources.srcDir 'src/rider/main/resources'
+    }
+}
+
+compileKotlin {
+    kotlinOptions { jvmTarget = "1.8" }
+}
+
+task compileDotNet {
+    doLast {
+        exec {
+            executable "dotnet"
+            args "msbuild","/t:Restore;Rebuild","${DotnetSolution}","/p:Configuration=${BuildConfiguration}"
+            workingDir rootDir
+        }
+    }
+}
+
+buildPlugin {
+    outputs.upToDateWhen { false }
+    doLast {
+        copy {
+            from "${buildDir}/distributions/${rootProject.name}-${version}.zip"
+            into "${rootDir}/output"
+        }
+
+        def changelogText = file("${rootDir}/CHANGELOG.md").text
+        def changelogMatches = changelogText =~ /(?s)(-.+?)(?=##|$)/
+        def changeNotes = changelogMatches.collect {
+            it[1].replaceAll(/(?s)- /, "\u2022 ").replaceAll(/`/, "").replaceAll(/,/, "%2C")
+        }.take(1).join("")
+
+        exec {
+            executable "dotnet"
+            args "msbuild","/t:Pack","${DotnetSolution}","/p:Configuration=${BuildConfiguration}","/p:PackageOutputPath=${rootDir}/output","/p:PackageReleaseNotes=${changeNotes}","/p:PackageVersion=${version}"
+        }
+    }
+}
+
+intellij {
+    type = 'RD'
+    version = "${ProductVersion}"
+    downloadSources = false
+    instrumentCode = false
+}
+
+apply plugin: 'com.jetbrains.rdgen'
+
+rdgen {
+    def modelDir = new File(rootDir, "protocol/src/main/kotlin/model")
+    def csOutput = new File(rootDir, "src/dotnet/${DotnetPluginId}/Rider")
+    def ktOutput = new File(rootDir, "src/rider/main/kotlin/xavalon/plugins/${RiderPluginId.replace('.','/').toLowerCase()}")
+
+    verbose = true
+    classpath "${rdLibDirectory}/rider-model.jar"
+    sources "${modelDir}/rider"
+    hashFolder = "${buildDir}"
+    packages = "model.rider"
+
+    generator {
+        language = "kotlin"
+        transform = "asis"
+        root = "com.jetbrains.rider.model.nova.ide.IdeRoot"
+        namespace = "com.jetbrains.rider.model"
+        directory = "$ktOutput"
+    }
+
+    generator {
+        language = "csharp"
+        transform = "reversed"
+        root = "com.jetbrains.rider.model.nova.ide.IdeRoot"
+        namespace = "JetBrains.Rider.Model"
+        directory = "$csOutput"
+    }
+}
+
+patchPluginXml {
+    def changelogText = file("${rootDir}/CHANGELOG.md").text
+    def changelogMatches = changelogText =~ /(?s)(-.+?)(?=##|$)/
+
+    changeNotes = changelogMatches.collect {
+        it[1].replaceAll(/(?s)\r?\n/, "<br />\n")
+    }.take(1).join('')
+}
+
+// buildSearchableOptions.onlyIf { false }
+
+prepareSandbox {
+    dependsOn compileDotNet
+
+    def outputFolder = "${rootDir}/src/dotnet/${DotnetPluginId}/bin/${DotnetPluginId}.Rider/${BuildConfiguration}"
+    def dllFiles = [
+            "$outputFolder/${DotnetPluginId}.dll",
+            "$outputFolder/${DotnetPluginId}.pdb",
+            "$outputFolder/XamlStyler.dll",
```

Comment:
This array describes the dll/pdbs to pack with the plugin. Ideally, plugins have no external dll's apart from their own, to avoid conflicts with other assemblies that have the same name. With `XamlStyler.dll` and `Irony.dll`, I think we're good in terms of potential conflicts.